route/otlp_trace_test.go

```diff
@@ -666,6 +666,124 @@ func TestOTLPHandler(t *testing.T) {
 		assert.Equal(t, "my-user-agent", event.Data.MetaRefineryIncomingUserAgent)
 	})
 
+	t.Run("postOTLPTrace error cases", func(t *testing.T) {
+		req := &collectortrace.ExportTraceServiceRequest{
+			ResourceSpans: []*trace.ResourceSpans{{
+				ScopeSpans: []*trace.ScopeSpans{{
+					Spans: helperOTLPRequestSpansWithStatus(),
+				}},
+			}},
+		}
+
+		t.Run("unsupported content type", func(t *testing.T) {
+			body, err := proto.Marshal(req)
+			require.NoError(t, err)
+
+			request, _ := http.NewRequest("POST", "/v1/traces", bytes.NewReader(body))
+			request.Header = http.Header{}
+			request.Header.Set("content-type", "application/xml") // Unsupported content type
+			request.Header.Set("x-honeycomb-team", legacyAPIKey)
+			request.Header.Set("x-honeycomb-dataset", "dataset")
+
+			w := httptest.NewRecorder()
+			router.postOTLPTrace(w, request)
+
+			assert.Equal(t, http.StatusUnsupportedMediaType, w.Code)
+			assert.Contains(t, w.Body.String(), "unsupported content-type")
+		})
+
+		t.Run("malformed protobuf body", func(t *testing.T) {
+			// Send invalid protobuf data
+			invalidBody := []byte("this is not valid protobuf data")
+
+			request, _ := http.NewRequest("POST", "/v1/traces", bytes.NewReader(invalidBody))
+			request.Header = http.Header{}
+			request.Header.Set("content-type", "application/protobuf")
+			request.Header.Set("x-honeycomb-team", legacyAPIKey)
+			request.Header.Set("x-honeycomb-dataset", "dataset")
+
+			w := httptest.NewRecorder()
+			router.postOTLPTrace(w, request)
+
+			assert.Equal(t, http.StatusInternalServerError, w.Code)
```

Comment:
🤔 Hmm. We ought to be returning HTTP errors in the 4xx range for these cases where the content is malformed or of a type we do not support. These are not 5xx (our fault) errors.